### DISCUSSION TITLE
Change panOnDrag with custom key

### DISCUSSION THREAD

**Discussion by jigrain** [xyflow/xyflow#4771] [Q&A] [answered]
There is an example in the official documentation https://reactflow.dev/learn/concepts/the-viewport
But this example of changing the control is a built-in alternative, and I haven't found any more information about whether I can somehow change the keys that pan. For example, I want to make panning possible only when shift is held down, can I do it?

**Comment by bcakmakoglu** [comment, marked as answer]
[What you could do is something like this.](https://codesandbox.io/p/sandbox/awesome-sutherland-6ffw2c?file=%2Fsrc%2FApp.tsx)

Basically, track if the `Shift` key is pressed and toggle `panOnDrag` if it is. 
Also probably disable or reassign the `selectionKeyCode` since that already one uses `Shift` by default.

```tsx
const isShiftKeyPressed = useKeyPress('Shift')

<ReactFlow
      nodes={nodes}
      onNodesChange={onNodesChange}
      edges={edges}
      onEdgesChange={onEdgesChange}
      panOnDrag={isShiftPressed}
      selectionKeyCode={null}
    >
```

**Reply by jigrain** [reply]
Thank you so much, I was paralleling my search and couldn't believe no one had asked about something like this in all that time. I hope your answer can help many more people.
